### PR TITLE
Pull config from env vars.

### PR DESCRIPTION
**Pull request by rlander**
12-Factor based settings to manage env vars.

### CODE REVIEW

**Review by smaniotto — COMMENTED** [review]
There's a mention to dj-database-url on the README file, can you update that part -- probably mentioning about different env files?

File: project_name/local.example.env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+DJANGO_DEBUG=True
```
Comment (jonatasbaldin) [inline project_name/local.example.env:1]: What do you think about creating a folder called `contrib` on the root to store files like these? We're probably going to have more "example files", so maybe it's good to centralized them!

Comment (jonatasbaldin) [reply project_name/local.example.env:1]: Also, I think we should keep comments about these variables, on what they mean and the pattern, something like:

```
# Toggles Django Debug mode
DJANGO_DEBUG=True

# Connection database string. E.g: provider://user:password@host:port/database
DATABASE_URL=postgres://postgres:postgres@db:5432/postgres
```
I know that for these two they are kinda silly, but once we have more stuff involved, it'll be awesome!

What do you think?

File: project_name/settings.py [hunk 2]
```diff
@@ -11,20 +11,30 @@
...
 import os
+from os.path import dirname, join, exists, abspath
+
+import environ  
+
+# Load operating system env variables and prepare to use them
+env = environ.Env()
+
+# .env file, should load only in development environment
+env_file = join(dirname(__file__), 'local.env')
```
Comment (jonatasbaldin) [inline project_name/settings.py:22]: Does it just get the vars from `local.env`? I think people are more used to just `.env`.

File: project_name/settings.py [hunk 3]
```diff
@@ -11,20 +11,30 @@
...
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
-BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+BASE_DIR = dirname(dirname(abspath(__file__)))
 
 # Quick-start development settings - unsuitable for production
 # See https://docs.djangoproject.com/en/1.10/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = '8#ubdv*jh_1u(6m4)^s^*pdo!&y_#jz)vv%5cp%8^*&%ztttxq'
+SECRET_KEY = env('DJANGO_SECRET_KEY', default='8#ubdv*jh_1u(6m4)^s^*pdo!&y_#jz)vv%5cp%8^*&%ztttxq')
```
Comment (jonatasbaldin) [inline project_name/settings.py:34]: I don't think we should have a default `SECRET_KEY`. Even locally, the developer should set its own string. Maybe we should leave without a default value.

What do you think?

**Review by jonatasbaldin — COMMENTED** [review]
Looking awesome! Just left a few thoughts 💃

File: project_name/settings.py (hunk 2 above)
Comment (rlander) [reply project_name/settings.py:22]: Well, I believe production servers should never rely on .env files, only on actual environment variables, so I use `local.env` to hammer home the idea.  But I'm totally cool with just `.env` too.

File: project_name/settings.py (hunk 3 above)
Comment (rlander) [reply project_name/settings.py:34]: I agree! I was just following the original template.

File: project_name/local.example.env (hunk 1 above)
Comment (rlander) [reply project_name/local.example.env:1]: I think both are very good ideas! Should these be separate issues?

Comment (jonatasbaldin) [reply project_name/local.example.env:1]: Sure!

File: project_name/settings.py (hunk 2 above)
Comment (jonatasbaldin) [reply project_name/settings.py:22]: I agree, whatever you think the best :)

File: project_name/settings.py (hunk 3 above)
Comment (jonatasbaldin) [reply project_name/settings.py:34]: No problemo! But I still think that the `default` should be empty 🤐

**Review by smaniotto — APPROVED** [review]

**Pull request merged**